src/controller/todo.controller.ts

```diff
@@ -50,16 +58,24 @@ export class TodosController {
     }
   }
   @Put(':id')
-  async update(@Param('id') id: number, @Body() data: CreateTodoDto, @Res() res: Response) {
+  async update(
+    @Param('id') id: number, @Body() data: CreateTodoDto, @Res() res: Response) {
+    const validFields: Array<keyof CreateTodoDto> = ['name', 'complete'];
+    const invalidFields = Object.keys(data).filter(campo => !validFields.includes(campo as keyof CreateTodoDto));
```

Comment:
estou notando que esse código se repete duas vezes é isso mesmo?

Reply:
Pergunta pro Adriano o que é NSR

Reply:
ele já tinha me ensinado, eu pensei sobre quando vi que tava repetindo, mas não tinha feito auhsasuhas crying is free